app/src/main/java/com/sunasterisk/travelapp/ui/list/restaurant/FilterDialogFragment.kt

```diff
@@ -1,17 +1,143 @@
+    private var maxPrice = 500
+    private var minRating = 3
+    private lateinit var listener: OnDataChange
+    private var viewFilter: View? = null
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = inflater.inflate(R.layout.dialog_filtter_restaurant, container, false)
+    ): View? {
+        viewFilter = inflater.inflate(R.layout.dialog_filtter_restaurant, container, false)
+        getArgument()
+        initView()
+        return viewFilter
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setContent()
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is OnDataChange) listener = context
+    }
+
+    override fun onCheckedChanged(group: RadioGroup?, checkedId: Int) {
+        sortByPrice = checkedId == R.id.radioButtonPrice
+    }
+
+    override fun onClick(view: View?) {
+        when (view?.id) {
+            R.id.textViewApply -> {
+                maxPrice = seekBarPrice.progress
+                minRating = seekBarRating.progress
+                listener.sendData(sortByPrice, maxPrice, minRating)
+                dismiss()
+            }
+            R.id.textViewReset -> {
+                sortByPrice = true
+                maxPrice = priceDefault
+                minRating = ratingDefault
+                setContent()
+            }
+        }
+    }
+
+    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+        if (seekBar?.id == R.id.seekBarPrice) {
+            textViewCurrentPrice.text =
+                resources.getString(R.string.title_current_price, progress)
+        } else {
+            textViewCurrentRating.text =
+                resources.getString(R.string.title_current_rating, progress)
+        }
+    }
+
+    override fun onStartTrackingTouch(p0: SeekBar?) {
+    }
+
+    override fun onStopTrackingTouch(p0: SeekBar?) {
+    }
+
+    private fun getArgument() {
+        arguments?.apply {
+            if (containsKey(ARGUMENT_SORT_BY)) {
+                sortByPrice = getBoolean(ARGUMENT_SORT_BY)
+            }
+            if (containsKey(ARGUMENT_MIN_RATING)) {
+                minRating = getInt(ARGUMENT_MIN_RATING)
+            }
+            if (containsKey(ARGUMENT_MAX_PRICE)) {
+                maxPrice = getInt(ARGUMENT_MAX_PRICE)
+            }
+        }
+    }
+
+    private fun initView() {
+        dialog?.setCanceledOnTouchOutside(false)
+        viewFilter?.let {
+            it.radioGroupSort.setOnCheckedChangeListener(this)
+            it.textViewApply.setOnClickListener(this)
+            it.textViewReset.setOnClickListener(this)
+            it.seekBarPrice.setOnSeekBarChangeListener(this)
+            it.seekBarRating.setOnSeekBarChangeListener(this)
+        }
```

Comment:
Dùng `apply` sẽ ổn hơn vì không có `it`

Reply:
em setOnlick cho các view ở dialogFragment này cũng thử apply rồi nhưng nếu dùng apply thì các view kia sẽ bị null